spec/integration/aws_internet_gateway_spec.rb

```diff
@@ -0,0 +1,46 @@
+require 'spec_helper'
+
+describe Chef::Resource::AwsInternetGateway do
+  extend AWSSupport
+
+  when_the_chef_12_server "exists", organization: 'foo', server_scope: :context do
+    with_aws "with a VPC and an internet gateway" do
+      vpc = nil
+      internet_gateway = nil
+
+      before {
```

Comment:
Hmm, it should be destroying the resource - do you get any messages about `failure during after block` or something like that from RSpec?  [This](https://github.com/chef/chef-provisioning-aws/blob/master/spec/aws_support.rb#L138-L156) is the logic for setting up the resource for create/destroy in the before/after blocks.